src/sentry/eventstore/snuba/backend.py

```diff
@@ -57,6 +57,43 @@ def get_events(
         Get events from Snuba.
         """
         assert filter, "You must provide a filter"
+
+        # This is an optimization for the Group.filter_by_event_id query where we
+        # have a single event ID and want to check all accessible projects for a
+        # direct hit. In this case it's usually faster to go to nodestore first.
+        if (
+            filter.event_ids
+            and filter.project_ids
+            and len(filter.event_ids) * len(filter.project_ids) < limit
+        ):
+            event_list = [
+                Event(project_id=project_id, event_id=event_id)
+                for event_id in filter.event_ids
+                for project_id in filter.project_ids
+            ]
+            self.bind_nodes(event_list)
+
+            events = [event for event in event_list if len(event.data)]
+            event_ids = {event.event_id for event in events}
+            project_ids = {event.project_id for event in events}
+            start = min(event.datetime for event in events)
+            end = max(event.datetime for event in events) + timedelta(seconds=1)
+
+            # TODO: skip the query for transactions
```

Comment:
If you use the discover dataset you'd be able to get transactions as well.

Reply:
This was just a note to self, since I left this half completed last night so probably isn't very clear for anyone else. The idea behind this is that we never go to Snuba for transactions since we can fetch the data from nodestore. group_id is the only field we can't get from nodestore and transactions don't have it.